src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php

```diff
@@ -206,7 +206,21 @@ protected function getConnection($connection = null)
      */
     protected function getTable($table)
     {
-        return is_subclass_of($table, Model::class) ? (new $table)->getTable() : $table;
+        $entity = $this->getModelEntity($table);
+
+        return $entity ? $entity->getTable() : $table;
+    }
+
+    protected function getTableConnection($table)
+    {
+        $entity = $this->getModelEntity($table);
+
+        return $entity ? $entity->getConnectionName() : null;
+    }
```

Comment:
9.x targets PHP 8.0+ so the nullsafe operator can be used in these two methods to avoid temporary variables:

```
return $this->getModelEntity($table)?->getConnectionName();
```